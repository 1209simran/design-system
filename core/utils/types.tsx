export type BaseProps = {
  className?: string,
  'data-test'?: string,
};

const baseProps = [
  'className',
  'data-test'
];

export const extractBaseProps = (props: Record<string, any>) => {
  const basePropsObj = baseProps.reduce((acc, curr) => {
    return (
      props[curr] ? { ...acc, [curr]: props[curr] } : { ...acc }
    );
  }, {});

  return basePropsObj;
};
