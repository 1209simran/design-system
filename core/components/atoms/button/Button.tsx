import * as React from 'react';
import Spinner from '@/components/atoms/spinner';
import Icon from '@/components/atoms/icon';
import classNames from 'classnames';

export type Appearance = 'basic' | 'primary' | 'success' | 'alert' | 'transparent';

export type Size = 'tiny' | 'regular' | 'large';

export type Alignment = 'left' | 'right';

export interface ButtonProps {
  /**
   * The size of `Button`
   * @default "regular"
   */
  size?: Size;
  /**
   * Color of the `Button`
   * @default "basic"
   */
  appearance?: Appearance;
  /**
   * Disables the `Button`, making it unable to be pressed
   */
  disabled?: boolean;
  /**
   * Expands the `Button` to full width
   */
  expanded?: boolean;
  /**
   * Adds loader inside `Button` when waiting for an action to complete
   */
  loading?: boolean;
  /**
   * Name of icon that is to be added inside `Button`
   * Material icon name
   */
  icon?: string;
  /**
   * Align icon left or right
   * @default "left"
   */
  iconAlign?: Alignment;
  /**
   * Text to be added inside `Button`
   */
  children?: string;
  /**
   * Specifies tab index of `Checkbox`
   */
  tabIndex?: number;
  /**
   * Adds className to `Card` component
   */
  className?: string;
  /**
   * Handler to be called when `Button` is clicked
   */
  onClick?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  /**
   * Handler to be called when mouse pointer enters `Button`.
   */
  onMouseEnter?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  /**
   * Handler to be called when mouse pointer leaves `Button`.
   */
  onMouseLeave?: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

const sizeMapping = {
  tiny: 12,
  regular: 16,
  large: 20,
};

export const Button = React.forwardRef<HTMLButtonElement, ButtonProps>((props, ref) => {
  const {
    appearance = 'basic',
    size = 'regular',
    iconAlign = 'left',
    tabIndex = 0,
    children,
    icon,
    expanded,
    loading,
    disabled,
    className,
    ...rest
  } = props;

  const buttonClass = classNames({
    ['Button']: true,
    ['Button--expanded']: expanded,
    [`Button--${size}`]: size,
    ['Button--square']: !children,
    [`Button--${appearance}`]: appearance,
    [`Button--iconAlign-${iconAlign}`]: children && iconAlign,
    [`${className}`]: className
  });

  const iconClass = classNames({
    ['Button-icon']: true,
    [`Button-icon--${iconAlign}`]: children && iconAlign
  });

  const spinnerClass = classNames({
    ['Button-spinner']: true,
    [`Button-spinner--${iconAlign}`]: children && iconAlign
  });

  return (
    <button ref={ref} className={buttonClass} disabled={disabled || loading} tabIndex={tabIndex} {...rest} >
      {loading && (
        <span className={spinnerClass}>
          <Spinner size="small" appearance={(appearance === 'basic' || appearance === 'transparent') ? 'secondary' : 'white'} />
        </span>
      )}
      {icon && !loading && (
        <div className={iconClass}>
          <Icon
            name={icon}
            appearance={disabled ? 'disabled' : (appearance === 'basic' || appearance === 'transparent') ? 'default' : 'white'}
            size={sizeMapping[size]}
          />
        </div>
      )}
      {children && `${children.charAt(0).toUpperCase()}${children.slice(1)}`}
    </button>
  );
});

Button.displayName = 'Button';

export default Button;
