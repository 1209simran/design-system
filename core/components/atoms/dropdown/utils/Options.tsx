export let dropdownOptions: any[] = [];
export const group1: any[] = [];
const group2: any[] = [];

for (let i = 1; i <= 40; i++) {
  group1.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

for (let i = 41; i <= 50; i++) {
  group2.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
  });
}

dropdownOptions.push({
  group: true,
  label: 'Group 1',
  items: group1,
});

dropdownOptions.push({
  group: true,
  label: 'Group 2',
  items: group2,
});

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option ${i}`,
    value: `Option ${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = group1.slice(0, 10);

export const storyWrapOptions = [
  {
    label: 'Design System Dropdown',
    value: 'Design System Dropdown'
  },
  {
    label: 'UI Kit Dropdown',
    value: 'UI Kit Dropdown',
  },
  {
    label: 'Innovaccer Analytics',
    value: 'Innovaccer Analytics'
  }
];

export const subInfoItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option 1',
    value: 'Option 1',
    icon: 'events'
  },
  {
    label: 'Option 2',
    value: 'Option 2',
    icon: 'events',
  },
  {
    label: 'Option 3',
    value: 'Option 3',
    icon: 'events',
  },
];
