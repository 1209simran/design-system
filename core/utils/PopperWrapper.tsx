import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { Manager, Reference, Popper } from 'react-popper';

type PositionType =
  | 'auto-start'
  | 'auto'
  | 'auto-end'
  | 'top-start'
  | 'top'
  | 'top-end'
  | 'right-start'
  | 'right'
  | 'right-end'
  | 'bottom-end'
  | 'bottom'
  | 'bottom-start'
  | 'left-end'
  | 'left'
  | 'left-start'
  | undefined;

type actionType = 'click' | 'hover';

enum Offsets {
  Small = '2px',
  Medium = '4px',
  Large = '8px'
}

interface IProps {
  trigger: React.ReactElement<any>;
  placement: PositionType;
  children: React.ReactElement<any>;
  style: React.CSSProperties;
  appendToBody: boolean;
  on?: actionType;
  offset: keyof typeof Offsets;
  closeOnBackdropClick?: boolean;
  hoverable?: boolean;
}

interface IState {
  open: boolean;
  mouseLeaveDelay: number;
  mouseEnterDelay: number;
}

class PopperWrapper extends React.Component<IProps, IState> {
  private triggerRef: React.RefObject<HTMLElement>;
  private popupRef: React.RefObject<HTMLDivElement>;
  private _timer?: number;

  constructor(props: IProps) {
    super(props);

    this.state = {
      open: false,
      mouseLeaveDelay: 50,
      mouseEnterDelay: 0
    };

    this.triggerRef = React.createRef();
    this.popupRef = React.createRef();
  }

  public componentWillUnmount() {
    clearTimeout(this._timer);
    document.removeEventListener('mousedown', this.doesNodeContainClick);
  }

  public handleMouseLeave = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: false });
        if (this.props.children.props.onMouseLeave) {
          this.props.children.props.onMouseLeave(event);
        }
      }, this.state.mouseLeaveDelay);
    } else {
      this.setState({ open: false });
      if (this.props.children.props.onMouseLeave) {
        this.props.children.props.onMouseLeave(event);
      }
    }
  }

  public handleMouseEnter = (event: React.MouseEvent<HTMLElement>) => {
    const { hoverable = false } = this.props;

    if (hoverable) {
      clearTimeout(this._timer);
      this._timer = window.setTimeout(() => {
        this.setState({ open: true });
        if (this.props.children.props.onMouseEnter) {
          this.props.children.props.onMouseEnter(event);
        }
      }, this.state.mouseEnterDelay);
    } else {
      this.setState({ open: true });
      if (this.props.children.props.onMouseEnter) {
        this.props.children.props.onMouseEnter(event);
      }
    }
  }

  public togglePopper = () => {
    this.setState({ open: !this.state.open });
  }

  public doesNodeContainClick = (event: Event) => {
    if (
      !(
        this.findDOMNode(this.popupRef).contains(event.target as HTMLElement) ||
        this.findDOMNode(this.triggerRef).contains(event.target as HTMLElement)
      )
    ) {
      this.togglePopper();
    }
  }

  public componentDidMount() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public componentDidUpdate() {
    const { on = 'click', closeOnBackdropClick = true } = this.props;
    const { open } = this.state;

    if (on === 'click' && open && closeOnBackdropClick) {
      document.addEventListener('mousedown', this.doesNodeContainClick);
    } else if (on === 'click' && !open && closeOnBackdropClick) {
      document.removeEventListener('mousedown', this.doesNodeContainClick);
    }
  }

  public getTriggerElement(trigger: React.ReactElement<any>, ref: React.Ref<any>, on: actionType) {
    const options =
      on === 'hover'
        ? {
          ref,
          onMouseEnter: this.handleMouseEnter,
          onMouseLeave: this.handleMouseLeave
        }
        : {
          ref,
          onClick: this.togglePopper
        };

    const element = React.cloneElement(
      (
        <span className="d-inline-block">
          {trigger}
        </span>
      ),
      options
      );

    return element;
  }

  public getChildrenElement(
    children: React.ReactElement<any>,
    ref: React.Ref<any>,
    placement: string,
    style: React.CSSProperties
  ) {
    const options = this.props.on === 'hover' ? {
      ref,
      style,
      onMouseEnter: this.handleMouseEnter,
      onMouseLeave: this.handleMouseLeave,
      'data-placement': placement
    } : {
      ref,
      style,
      'data-placement': placement
    };

    const element = React.cloneElement(children, options);
    return element;
  }

  public render() {
    const { trigger, children, placement, appendToBody, on = 'click', offset } = this.props;
    const { open } = this.state;

    return (
      <Manager>
        <Reference innerRef={this.triggerRef}>{({ ref }) => this.getTriggerElement(trigger, ref, on)}</Reference>
        {open &&
          appendToBody &&
          ReactDOM.createPortal(
            (
              /* tslint:disable:no-shadowed-variable */
              <Popper placement={placement} innerRef={this.popupRef}>
                {({ ref, style, placement }) => {
                  const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
                  return this.getChildrenElement(children, ref, placement, newStyle);
                }}
              </Popper>
            ),
            document.body
          )}
        {open && !appendToBody && (
          <Popper placement={placement} innerRef={this.popupRef}>
            {({ ref, style, placement }) => {
              const newStyle = offset ? this.getUpdatedStyle(style, placement, offset) : style;
              return this.getChildrenElement(children, ref, placement, newStyle);
            }}
          </Popper>
        )}
      </Manager>
    );
  }

  private getUpdatedStyle = (
    oldStyle: React.CSSProperties,
    placement: PositionType,
    offset: keyof typeof Offsets = 'Medium'
  ) => {
    const { style = {} } = this.props;
    const newStyle = { ...style, ...oldStyle };
    const position = placement ? placement.split('-')[0] : placement;
    switch (position) {
      case 'top':
        newStyle.marginBottom = Offsets[offset];
        break;

      case 'bottom':
        newStyle.marginTop = Offsets[offset];
        break;

      case 'left':
        newStyle.marginRight = Offsets[offset];
        break;

      case 'right':
        newStyle.marginLeft = Offsets[offset];
        break;
    }
    return newStyle;
  }

  private findDOMNode = (ref: React.RefObject<HTMLElement>) => {
    return ReactDOM.findDOMNode(ref.current!) as HTMLElement;
  }
}

export default PopperWrapper;
