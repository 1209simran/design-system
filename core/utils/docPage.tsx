// @ts-nocheck

import * as React from 'react';
import { Title, Props, Description, Primary, Story, PropsProps } from '@storybook/addon-docs/blocks';

interface DocPageProps {
  title?: string;
  description?: React.ReactNode;
  props?: PropsProps;
}

export const docPage = (props: DocPageProps) => {
  const {
    title,
    description,
    props,
    stories
  } = props;

  const separatorIndex = title?.indexOf('|');

  return (
    <>
      <Title>{title && separatorIndex !== -1 ? title.slice(separatorIndex + 1) : title}</Title>
      <Description>
        {description}
      </Description>
      {/* <Primary /> */}
      {/* <Story id="switch-state--alert"  /> */}
      <Props {...props} />
    </>
  );
};

export default docPage;
