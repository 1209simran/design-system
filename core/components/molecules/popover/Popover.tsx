import * as React from 'react';
import classNames from 'classnames';
import { PopperWrapper } from '@/utils';
import { BaseProps } from '@/utils/types';

export type Position = 'top' | 'top-start' | 'top-end' | 'bottom' | 'bottom-start' | 'bottom-end';
type ActionType = 'click' | 'hover';

export interface PopoverProps extends BaseProps {
  /**
   *  Position to place the `trigger`
   * @default "bottom"
   */
  position?: Position;
  /**
   * Appends `trigger` wrapper inside body
   * @default true
   */
  appendToBody?: boolean;
  /**
   * Vertical offset from `trigger`
   */
  verticalOffset?: number;
  /**
   * Element triggering the `Popover`
   */
  trigger: React.ReactElement<any>;
  /**
   * Holds `Popover` on hover
   */
  hoverable?: boolean;
  /**
   * Changes background of `Popover`
   */
  dark?: boolean;
  /**
   * Closes `Popover` on outside click
   * @default true
   */
  closeOnBackdropClick?: boolean;
  /**
   * Event triggering the `Popover`
   * @default 'click'
   */
  on?: ActionType;
  /**
   * Handles open/close
   */
  open?: boolean;
  /**
   * Adds CSS to `Popover` element
   * @default {}
   */
  style?: React.CSSProperties;
  /**
   * Callback after `Popover` is toggled
   */
  onToggle?: (open: boolean, type?: string) => void;
  /**
   * To be rendered in `Popover` component
   */
  children: React.ReactNode;
}

export const Popover = (props: PopoverProps) => {
  const {
    position = 'bottom',
    closeOnBackdropClick = true,
    appendToBody = true,
    on = 'click',
    style = {},
    dark,
    hoverable,
    children,
    trigger,
    onToggle,
    className,
  } = props;

  const [open, setOpen] = React.useState<boolean>(props.open || false);
  React.useEffect(() => {
    if (onToggle) {
      if (props.open !== undefined) setOpen(props.open);
    }
  }, [props.open]);

  const onToggleFunction: PopoverProps['onToggle'] = newOpen => {
    setOpen(newOpen);
  };

  const classes = classNames({
    Popover: true,
    ['Popover--dark']: dark
  }, className);

  const PopoverWrapper = (
    <div className={classes} >
      {children}
    </div>
  );

  const popperOptions = {
    trigger,
    appendToBody,
    closeOnBackdropClick,
    on,
    hoverable,
    style,
    open,
    onToggle: onToggle || onToggleFunction,
    placement: position
  };

  return (
    <PopperWrapper {...popperOptions} offset="Large" >
      {PopoverWrapper}
    </PopperWrapper>
  );
};

Popover.displayName = 'Popover';

export default Popover;
