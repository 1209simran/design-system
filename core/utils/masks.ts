import { Mask } from '@/utils/types';
import { DatePickerProps } from "@/index.type";

type DateFormat = DatePickerProps['inputFormat'];
export const date: Record<DateFormat, Mask> = {
  'dd/mm/yyyy': [/[0123]/, /\d/, '/', /\[01]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'mm/dd/yyyy': [/[01]/, /\d/, '/', /[0123]/, /\d/, '/', /\d/, /\d/, /\d/, /\d/],
  'yyyy/mm/dd': [/\d/, /\d/, /\d/, /\d/, '/', /[01]/, /\d/, '/', /[0123]/, /\d/],
  'dd-mm-yyyy': [/[0123]/, /\d/, '-', /[01]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'mm-dd-yyyy': [/[01]/, /\d/, '-', /[0123]/, /\d/, '-', /\d/, /\d/, /\d/, /\d/],
  'yyyy-mm-dd': [/\d/, /\d/, /\d/, /\d/, '-', /[01]/, /\d/, '-', /[0123]/, /\d/],
};
