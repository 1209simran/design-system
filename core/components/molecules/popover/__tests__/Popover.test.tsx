import * as React from 'react';
import { shallow } from 'enzyme';
import Popover, { IPopoverProps as IProps } from '../Popover';
import Button from '@/components/atoms/button';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';

const position = ['top', 'top-start', 'top-end', 'bottom', 'bottom-start', 'bottom-end'];
const BooleanValue = [true, false];
const NumberValue = 10;
const Style = {
  width: valueHelper('150px', { required: true }),
  height: valueHelper('150px', { required: true })
};
const ReactElementValue = <Button appearance="basic">Open Popup</Button>;

describe('Popover component', () => {
  const mapper = {
    verticalOffset: valueHelper(NumberValue, { required: true }),
    trigger: valueHelper(ReactElementValue, { required: true }),
    dark: valueHelper(BooleanValue, { required: true, iterate: true }),
    closeOnBackdropClick: valueHelper(BooleanValue, { required: true, iterate: true }),
    style: valueHelper(Style, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Popover
          {...attr}
        >
          Popover
        </Popover>
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('Popover component', () => {
  const mapper = {
    position: valueHelper(position, { required: true, iterate: true }),
    verticalOffset: valueHelper(NumberValue, { required: true }),
    trigger: valueHelper(ReactElementValue, { required: true }),
    closeOnBackdropClick: valueHelper(BooleanValue, { required: true, iterate: true }),
    style: valueHelper(Style, { required: true }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as IProps;

    it(testMessageHelper(attr), () => {
      const tree = shallow(
        <Popover
          {...attr}
        >
          Popover
        </Popover>
      );
      expect(tree).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});
