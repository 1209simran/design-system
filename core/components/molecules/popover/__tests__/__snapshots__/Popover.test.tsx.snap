// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component 
  position: "bottom", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: false, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={false}
  offset="Large"
  on="click"
  placement="bottom"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "bottom", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: true, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={true}
  offset="Large"
  on="click"
  placement="bottom"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "bottom-end", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: false, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={false}
  offset="Large"
  on="click"
  placement="bottom-end"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "bottom-end", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: true, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={true}
  offset="Large"
  on="click"
  placement="bottom-end"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "bottom-start", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: false, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={false}
  offset="Large"
  on="click"
  placement="bottom-start"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "bottom-start", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: true, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={true}
  offset="Large"
  on="click"
  placement="bottom-start"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "top", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: false, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={false}
  offset="Large"
  on="click"
  placement="top"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "top", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: true, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={true}
  offset="Large"
  on="click"
  placement="top"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "top-end", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: false, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={false}
  offset="Large"
  on="click"
  placement="top-end"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "top-end", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: true, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={true}
  offset="Large"
  on="click"
  placement="top-end"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "top-start", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: false, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={false}
  offset="Large"
  on="click"
  placement="top-start"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  position: "top-start", verticalOffset: 10, trigger: "[ReactNode]", closeOnBackdropClick: true, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={true}
  offset="Large"
  on="click"
  placement="top-start"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  verticalOffset: 10, trigger: "[ReactNode]", dark: false, closeOnBackdropClick: false, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={false}
  offset="Large"
  on="click"
  placement="bottom"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  verticalOffset: 10, trigger: "[ReactNode]", dark: false, closeOnBackdropClick: true, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={true}
  offset="Large"
  on="click"
  placement="bottom"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  verticalOffset: 10, trigger: "[ReactNode]", dark: true, closeOnBackdropClick: false, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={false}
  offset="Large"
  on="click"
  placement="bottom"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover Popover--dark"
  >
    Popover
  </div>
</PopperWrapper>
`;

exports[`Popover component 
  verticalOffset: 10, trigger: "[ReactNode]", dark: true, closeOnBackdropClick: true, style: {"width":{"required":true,"value":"150px"},"height":{"required":true,"value":"150px"}}
 1`] = `
<PopperWrapper
  appendToBody={true}
  closeOnBackdropClick={true}
  offset="Large"
  on="click"
  placement="bottom"
  style={
    Object {
      "height": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
      "width": Object {
        "iterate": undefined,
        "required": true,
        "value": "150px",
      },
    }
  }
  trigger={
    <Button
      appearance="basic"
    >
      Open Popup
    </Button>
  }
>
  <div
    className="Popover Popover--dark"
  >
    Popover
  </div>
</PopperWrapper>
`;
